### PR TITLE
GreenField API: Configure Store Lightning Payment Method

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

**Review by dennisreimann — COMMENTED** [review]
Nice, I can use this for my LNbank setup automation script. Will test the PR in this context  later :)

File: BTCPayServer/Controllers/GreenField/StoreLightningNetworkPaymentMethodsController.cs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+            if (_lightningNetworkOptions.Value.InternalLightningByCryptoCode.TryGetValue(cryptoCode, out var connectionString))
+            {
+                return CanUseInternalLightning() ? connectionString : null;
+            }
+            return null;
+        }
+        
+        private bool CanUseInternalLightning()
+        {
+            return (_btcPayServerEnvironment.IsDeveloping || User.IsInRole(Roles.ServerAdmin) || _cssThemeManager.AllowLightningInternalNodeForAll);
```
[the remark this answers is not in the document]
Comment (Kukks) [reply BTCPayServer/Controllers/GreenField/StoreLightningNetworkPaymentMethodsController.cs:246]: technical debt, CssThemeManager is really just a cache for the PolicySettings at this point. It is ripe for refactoring

File: BTCPayServer/Controllers/GreenField/StoreLightningNetworkPaymentMethodsController.cs [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            [FromQuery] bool enabledOnly = false)
+        {
+            var blob = Store.GetStoreBlob();
+            var excludedPaymentMethods = blob.GetExcludedPaymentMethods();
+            return Ok(Store.GetSupportedPaymentMethods(_btcPayNetworkProvider)
+                .Where((method) => method.PaymentId.PaymentType == PaymentTypes.LightningLike)
+                .OfType<DerivationSchemeSettings>()
+                .Select(strategy =>
+                    new LightningNetworkPaymentMethodData(strategy.PaymentId.CryptoCode,
+                        strategy.AccountDerivation.ToString(), !excludedPaymentMethods.Match(strategy.PaymentId)))
```
Comment (dennisreimann) [inline BTCPayServer/Controllers/GreenField/StoreLightningNetworkPaymentMethodsController.cs:62]:
```suggestion
                .OfType<LightningSupportedPaymentMethod>()
                .Select(paymentMethod =>
                    new LightningNetworkPaymentMethodData(paymentMethod.PaymentId.CryptoCode,
                        paymentMethod.GetLightningUrl().ToString(), !excludedPaymentMethods.Match(paymentMethod.PaymentId)))
```

**Review by dennisreimann — CHANGES_REQUESTED** [review]
Two things I came across while testing.

**Review by dennisreimann — APPROVED** [review]
Ok, tested the other endpoints as well and this looks good to me. 👍

**Comment by NicolasDorier** [comment]
Reverted, need more review and also need for next feature release rather than bug fix release.

**Pull request merged**